lib/ecto/migration.ex

```diff
@@ -1358,16 +1361,24 @@ defmodule Ecto.Migration do
   end
 
   @doc false
-  def __primary_key__() do
-    case Runner.repo_config(:migration_primary_key, []) do
-      false ->
-        false
-
-      opts when is_list(opts) ->
-        opts = Keyword.put(opts, :primary_key, true)
-        {name, opts} = Keyword.pop(opts, :name, :id)
-        {type, opts} = Keyword.pop(opts, :type, :bigserial)
-        {name, type, opts}
+  def __primary_key__(table) do
+    case table.primary_key do
+      false -> false
+
+      true ->
+        case Runner.repo_config(:migration_primary_key, []) do
+          false -> false
+          opts when is_list(opts) -> pk_opts_to_tuple(opts)
+        end
+
+      opts when is_list(opts) -> pk_opts_to_tuple(opts)
```

Comment:
there's a slight difference in behaviour from the previous function: if the user supplies some kind of non-boolean value for this option it would previously ignore it and go for the repo config. now it will raise if it's not a  keyword list.

Reply:
Up to you whether that's important to stay consistent on given that it's outside of the documented possible values for the option. Happy to make it replicate the existing behaviour if needed though.

Reply:
I would say raising with a more specific message is best. That way it will help users understand quickly why it's not working anymore. And they won't think the invalid option they are passing in is doing anything.